src/phoenix/server/api/routers/v1/dataset_examples.py

```diff
@@ -0,0 +1,99 @@
+from sqlalchemy import func, select
+from starlette.responses import JSONResponse, Response
+from starlette.status import HTTP_404_NOT_FOUND
+from strawberry.relay import GlobalID
+
+from phoenix.db.models import Dataset, DatasetExample, DatasetExampleRevision, DatasetVersion
+
+
+async def list_dataset_examples(request):
+    dataset_id = GlobalID.from_id(request.path_params["id"])
+    raw_version_id = request.query_params.get("version")
+    version_id = GlobalID.from_id(raw_version_id) if raw_version_id else None
+
+    if (dataset_type := dataset_id.type_name) != "Dataset":
+        return Response(
+            content=f"ID {dataset_id} refers to a {dataset_type}", status_code=HTTP_404_NOT_FOUND
+        )
+
+    if version_id and (version_type := version_id.type_name) != "DatasetVersion":
+        return Response(
+            content=f"ID {version_id} refers to a {version_type}", status_code=HTTP_404_NOT_FOUND
+        )
```

Comment:
```suggestion
    if (dataset_type := dataset_id.type_name) != Dataset.__name__:
        return Response(
            content=f"ID {dataset_id} refers to a {dataset_type}", status_code=HTTP_404_NOT_FOUND
        )

    if version_id and (version_type := version_id.type_name) != DatasetVersion.__name__:
        return Response(
            content=f"ID {version_id} refers to a {version_type}", status_code=HTTP_404_NOT_FOUND
        )
```

Suggestion: We can make this robust to changes in the type name.